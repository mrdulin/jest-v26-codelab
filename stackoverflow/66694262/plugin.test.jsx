import { Plugins } from './plugin';
import { shallow } from 'enzyme';

const pluginsData = [{ plugin: 'Apps (Coming soon)' }];

const wrapper = shallow(<Plugins pluginsData={pluginsData} />);

describe('test home page', () => {
  it('should render home page successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct text in item', () => {
    expect(wrapper.find('.up-coming-text').get(0).props.children).toEqual('Apps (Coming soon)');
  });

  it('renders pluginsData length', () => {
    expect(wrapper.find('.up-coming-text')).toBeDefined();
    expect(wrapper.find('.up-coming-text').length).toEqual(pluginsData.length);
  });
});
