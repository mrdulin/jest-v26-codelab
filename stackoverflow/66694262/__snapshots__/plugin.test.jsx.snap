// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home page should render home page successfully 1`] = `
<ul
  className="plugin-list"
>
  <li
    appearance="subtle"
    className="item"
    onClick={[Function]}
    weight="medium"
  >
    Plugins
  </li>
  <li
    className="up-coming-text"
    key="0"
  >
    Apps (Coming soon)
  </li>
</ul>
`;
